src/main/java/seedu/eduke8/Parser.java

```diff
@@ -1,13 +1,27 @@
 package seedu.eduke8;
 
 
-import seedu.eduke8.quiz.QuizInterface;
+import seedu.eduke8.option.OptionInterface;
+import seedu.eduke8.question.QuestionInterface;
+import seedu.eduke8.quiz.Quiz;
+import seedu.eduke8.topic.TopicListInterface;
 import seedu.eduke8.ui.UiInterface;
 
+import java.util.ArrayList;
+
 public class Parser implements ParserInterface {
 
-    public void parseCommand(String userInput) {
+    public Parser() {
+    }
+
+    public void parseCommand(TopicListInterface topicList, String userInput) {
+        int numOfQuestions = 0;
+        String topicName = "";
         String[] commandArr = userInput.trim().split(" ", 0);
+        if (commandArr.length == 3 && commandArr[0].equals("quiz")) {
+            numOfQuestions = Integer.parseInt(commandArr[2].substring(2));
+            topicName = commandArr[1].substring(2);
+        }
         UiInterface ui = null;
```

Comment:
Remember to initialize this next time as well

Reply:
It was initialised above the if loop.

Reply:
sorry I mean to comment on ui, didn't know the top part got included